src/juicebox/tools/utils/original/Preprocessor.java

```diff
@@ -239,6 +245,46 @@ public void setRandomizePosition(boolean allowPositionsRandomization) {
         Preprocessor.allowPositionsRandomization = allowPositionsRandomization;
     }
 
+    private static FragmentCalculation findFragMap(List<FragmentCalculation> maps, String chr, int bp, int frag) {
+        //potential maps that this strand could come from
+        ArrayList<FragmentCalculation> mapsFound = new ArrayList<>();
+        for (FragmentCalculation fragmentCalculation : maps) {
+            int low = 1;
+            int high = 1;
+
+            if (frag > fragmentCalculation.getNumberFragments(chr)) {
+                // definitely not this restriction site file for certain
+                continue;
+            }
+            
+            try {
+                if (frag == 0) {
+                    high = fragmentCalculation.getSites(chr)[frag];
+                } else if (frag == fragmentCalculation.getNumberFragments(chr)) {
+                    high = fragmentCalculation.getSites(chr)[frag - 1];
+                    low = fragmentCalculation.getSites(chr)[frag - 2];
+                } else {
+                    high = fragmentCalculation.getSites(chr)[frag];
+                    low = fragmentCalculation.getSites(chr)[frag - 1];
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+                System.out.println(String.format("fragment: %d, number of frags: %d", frag, fragmentCalculation.getNumberFragments(chr)));
+
+            }
+
+            // does bp fit in this range?
+            if (bp >= low && bp <= high) {
+                mapsFound.add(fragmentCalculation);
+            }
+        }
+        if (mapsFound.size() == 1) {
+            return mapsFound.get(0);
+        }
```

Comment:
can you create a counter for the different types of fragments matching error? i.e. the number of read pairs with ambiguous (>1 match) that are not resolved and those with 0 matches?